base/operators.jl

```diff
@@ -1153,6 +1153,29 @@ julia> filter(!isletter, str)
 !(f::Function) = (!) ∘ f
 !(f::ComposedFunction{typeof(!)}) = f.inner #allows !!f === f
 
+"""
+    -f::Function
+
+When the argument of `-` is a function, it returns `(-) ∘ f`.
+
+See also [`!`](@ref) for boolean negation, and [`∘`](@ref).
+
+# Examples
+```jldoctest
+julia> map(-abs2, Tuple(-2:2))
+(-4, -1, 0, -1, -4)
+
+julia> sum(-log10, [1 0.1; 0.01 0.001]; dims=1)
+1×2 Matrix{Float64}:
+ 2.0  4.0
+```
+
+!!! compat "Julia 1.12"
+    This method requires at least Julia 1.12.
+"""
+-(f::Function) = (-) ∘ f
+-(f::ComposedFunction{typeof(-)}) = f.inner #allows -(-f) === f
```

Comment:
This method seems fit for an independent PR. I think it'd be merged easier than the feature addition.

Reply:
Happy to remove, it's just copied over from the `!!f` case. Where it allows this slightly odd result:
```julia
julia> (!!log)(2)
0.6931471805599453

julia> !!log(2)
ERROR: MethodError: no method matching !(::Float64)
```
although I doubt that ever matters in practice.